pkgs/servers/sql/postgresql/buildPostgresExtension.nix

```diff
@@ -0,0 +1,124 @@
+# time, pg_config only points to the PostgreSQL derivation's paths.
+#
+# When building extensions, the paths provided by pg_config are used for two
+# purposes:
+# - To find postgres libs and headers and reference those paths via -L and -I flags.
+# - To determine the correct install directory.
+#
+# The PGXS Makefiles also support an environment variable DESTDIR, which is added as
+# a prefix to all install locations. This is primarily used for temporary installs
+# while running the test suite. Since pg_config returns absolute paths to /nix/store
+# for us, using DESTDIR will result in install locations of the form:
+#   $DESTIDR/nix/store/<postgresql-output>/...
+#
+# In multiple iterations, the following approaches have been tried to work around all
+# of this:
+# 1. For a long time, all extensions in nixpkgs just overwrote the installPhase
+#    and moved the respective files to the correct location manually. This approach
+#    is not maintainable, because whenever upstream adds a new file, we'd have to
+#    make sure the file is correctly installed as well. Additionally, it makes adding
+#    a new extension harder than it should be.
+#
+# 2. A wrapper around pg_config could just replace the returned paths with paths to
+#    $out of currently building derivation, i.e. the extension. This works for install-
+#    ation, but breaks for any of the libs and headers the extension needs from postgres
+#    itself.
+#
+# 3. A variation of 2., but make the pg_config wrapper only return the changed paths
+#    during the installPahse. During configure and build, it would return the regular
+#    paths to the PostgreSQL derivation. This works better, but not for every case.
+#    Some extensions try to be smarter and search for the "postgres" binary to deduce
+#    the necessary paths from that. Those would still need special handling.
+#
+# 4. Use the fact that DESTDIR is prepended to every installation directory - and only
+#    there, to run a replacement of all Makefiles in postgres' lib/pgxs/ folder and
+#    all Makefiles in the extension's source. "$DESTDIR/$bindir" can be replaced with
+#    "$out/bin" etc. - thus mapping the installPhase directly into the right output.
+#    This works beautifully - for the majority of cases. But it doesn't work for
+#    some extensions that use CMake. And it doesn't work for some extensions that use
+#    custom variables instead of the default "bindir" and friends.
+#
+# 5. Just set DESTDIR to the extensions's output and then clean up afterward. This will
+#    result in paths like this:
+#      /nix/store/<extension-output>/nix/store/<postgresql-output>/...
+#    Directly after the installPhase, we'll move the files in the right folder.
+#    This seems to work consistently across all extensions we have in nixpkgs right now.
+#    Of course, it would break down for any extension that doesn't support DESTDIR -
+#    but that just means PGXS is not used either, so that's ok.
```

Comment:
```suggestion
#    but that just means PGXS is not used either, so that's OK.
```
or
```suggestion
#    but that just means PGXS is not used either, so that's okey.
```